tests/scripts/check_files.py

```diff
@@ -367,9 +367,13 @@ class LicenseIssueTracker(LineIssueTracker):
 
     heading = "License issue:"
 
+    TF_PSA_CRYPTO_PATH = "tf-psa-crypto/"
+    if build_tree.looks_like_tf_psa_crypto_root(os.getcwd()):
```

Comment:
Calling to the OS during the definition of the _class_, rather than in the constructor (i.e. when constructing an _object_), is a [code smell](https://en.wikipedia.org/wiki/Code_smell). I'm not sure if it breaks anything given how the script is written, but it's at least fragile.

Why should we bother to do this anyway? This is just to build a list of exceptions, so we can have both `'tf-psa-crypto/wibble'` and `'wibble'` in the exception list.

Reply:
Perhaps my understanding of the task was wrong. I interpreted it as "enable tests to run independently of Mbed TLS within TF-PSA-Crypto". Without adding a prefix to the path, running this script inside TF-PSA-Crypto, will fail as this directory cannot be found `tf-psa-crypto/drivers/(?=(everest)/.*)`.

Reply:
This isn't a path that needs to exist, it's a regular expression that matches paths. It's ok if we have `tf-psa-crypto/drivers/(?=(everest)/.*)` in the exclusion list in TF-PSA-Crypto (it will match nothing) and `drivers/(?=(everest)/.*)` in the exclusion list in mbedtls (it will match nothing).

Reply:
I believe I have understoond this now. I have currently changed the regex from `tf-psa-crypto/drivers/(?=(everest)/.*)` to `.*drivers/(?=(everest)/.*)`. This will now match any possible path containing `drivers/everest/` because without it, running the test inside TF-PSA-Crypto will fail.
```
cd tf-psa-crypto
../test/scripts/check_file.py
```

Reply:
I'm slightly uneasy at also matching e.g. `unrelated/mountain-road-drivers/everest/foo`. Maybe `(tf-psa-crypto/)?drivers/(?=(everest)/.*)` or something like that? Or at least `(?:\A|/)drivers/(?=(everest)/.*)` ?

Reply:
> Perhaps my understanding of the task was wrong. I interpreted it as "enable tests to run independently of Mbed TLS within TF-PSA-Crypto". 

Yes that's correct.